src/main/java/com/monumental/services/MonumentService.java

```diff
@@ -320,24 +333,49 @@ private void getRelatedRecords(List<Monument> monuments, String fieldName) {
 
         String capitalizedFieldName = fieldName.substring(0, 1).toUpperCase() + fieldName.substring(1);
 
-        Map<Integer, List> map = new HashMap<>();
-        for (Monument monument : monumentsWithRecords) {
-            try {
-                map.put(monument.getId(), (List) Monument.class.getDeclaredMethod("get" + capitalizedFieldName).invoke(monument));
-            } catch (NoSuchMethodException | InvocationTargetException | IllegalAccessException e) {
-                System.err.println("Invalid field name: " + fieldName);
-                System.err.println("Occurred while trying to use getter: get" + capitalizedFieldName);
-                e.printStackTrace();
+        // "monumentTags" is a special case because it's a Set, not a List
+        if (fieldName.equals("monumentTags")) {
+            Map<Integer, Set> map = new HashMap<>();
+            for (Monument monument : monumentsWithRecords) {
+                try {
+                    map.put(monument.getId(), (Set) Monument.class.getDeclaredMethod("get" + capitalizedFieldName).invoke(monument));
+                } catch (NoSuchMethodException | InvocationTargetException | IllegalAccessException e) {
+                    System.err.println("Invalid field name: " + fieldName);
+                    System.err.println("Occurred while trying to use getter: get" + capitalizedFieldName);
+                    e.printStackTrace();
+                }
+            }
+
+            for (Monument monument : monuments) {
+                try {
+                    Monument.class.getDeclaredMethod("set" + capitalizedFieldName, Set.class).invoke(monument, map.get(monument.getId()));
+                } catch (NoSuchMethodException | InvocationTargetException | IllegalAccessException e) {
+                    System.err.println("Invalid field name: " + fieldName);
+                    System.err.println("Occurred while trying to use setter: set" + capitalizedFieldName);
+                    e.printStackTrace();
+                }
```

Comment:
We could remove this special case since it's never used, but we'd need to re-add it in the future if we need it.